rmw_dds_common/src/qos.cpp

```diff
@@ -0,0 +1,243 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <sstream>
+
+#include "rmw/qos_profiles.h"
+
+#include "rmw_dds_common/qos.hpp"
+
+namespace rmw_dds_common
+{
+
+static bool
+operator==(rmw_time_t t1, rmw_time_t t2)
+{
+  return t1.sec == t2.sec && t1.nsec == t2.nsec;
+}
+
+static bool
+operator!=(rmw_time_t t1, rmw_time_t t2)
+{
+  return !(t1 == t2);
+}
+
+static bool
+operator<(rmw_time_t t1, rmw_time_t t2)
+{
+  if (t1.sec < t2.sec) {
+    return true;
+  } else if (t1.sec == t2.sec && t1.nsec < t2.nsec) {
+    return true;
+  }
+  return false;
+}
+
+static std::ostream &
+operator<<(std::ostream & out, const rmw_time_t & t)
+{
+  out << "{" << t.sec << "s " << t.nsec << "ns}";
+  return out;
+}
+
+static void
+_write_to_buffer(const std::string message, char * buffer, size_t buffer_size)
```

Comment:
```suggestion
_write_to_buffer(const char * message, char * buffer, size_t buffer_size)
```

the first argument can be a cstring.

L237 needs to change to:   `_write_to_buffer(reason_ss.str().c_str(), reason, reason_size);`

Reply:
Updated in 132b3e1